ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLTask.java

```diff
@@ -85,7 +86,24 @@ public int execute() {
       } else {
         throw new IllegalArgumentException("Unknown DDL request: " + ddlDesc.getClass());
       }
-    } catch (Throwable e) {
+    } catch(HiveException e){
+      int excpErrorCode = ReplUtils.handleException(work.isReplication(), e, work.getDumpDirectory(),
+              work.getMetricCollector(), getName(), conf);
+      if(excpErrorCode == 10241){
```

Comment:
what is this magic number, do we have constants for that?

Reply:
![image](https://github.com/apache/hive/assets/157357971/7f34470a-7adf-425d-83dd-edfbae1068cd)

Yes, it the Error code corresponding to "Table Not Partitioned" error we are handling, you can find the same in [https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/ql/ErrorMsg.java](url)

Reply:
then use it: `ErrorMsg.TABLE_NOT_PARTITIONED.getErrorCode()`